environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/freeipa/FreeIpaCreationHandler.java

```diff
@@ -307,14 +333,28 @@ private void setUseCcm(Tunnel tunnel, CreateFreeIpaRequest createFreeIpaRequest)
         createFreeIpaRequest.setTunnel(tunnel);
     }
 
-    private List<InstanceGroupRequest> createInstanceGroupRequests(SecurityGroupRequest securityGroupRequest, FreeIpaCreationDto freeIpaCreation) {
+    private List<InstanceGroupRequest> createInstanceGroupRequests(CreateFreeIpaRequest createFreeIpaRequest,
+        SecurityGroupRequest securityGroupRequest,
+        EnvironmentDto environment,
+        boolean multiAzRequired) {
         List<InstanceGroupRequest> instanceGroupRequests = new LinkedList<>();
+        FreeIpaCreationDto freeIpaCreation = environment.getFreeIpaCreation();
         InstanceGroupRequest instanceGroupRequest = new InstanceGroupRequest();
         instanceGroupRequest.setName(MASTER_GROUP_NAME);
         instanceGroupRequest.setNodeCount(freeIpaCreation.getInstanceCountByGroup());
         instanceGroupRequest.setType(InstanceGroupType.MASTER);
         instanceGroupRequest.setSecurityGroup(securityGroupRequest);
         instanceGroupRequest.setInstanceTemplateRequest(createInstanceTemplate(freeIpaCreation));
+        if (multiAzRequired && multiAzValidator.suportedMultiAzForEnvironment(environment.getCloudPlatform())) {
```

Comment:
What about the `else` clause? Is not it an error to ask for multi-AZ on a platform that is not supported?

Reply:
if not ssuported etthen we dont configure